### DISCUSSION TITLE
How to correctly use Suspense with NuxtPage?

### DISCUSSION THREAD

**Discussion by ilyaDegtyarenko** [nuxt/nuxt#16117] [Questions]
I don't see fallback content, just white screen

_Docs https://vuejs.org/guide/built-ins/suspense.html#combining-with-other-components_

**Parent component (1/2 option)**
```
<template>
    <div>
        <h1>Parent</h1>
        
        <Suspense>
            <NuxtPage/>

            <template #fallback>
                Loading...
            </template>
        </Suspense>
    </div>
</template>
```

**Parent component (2/2 option)**
```
<template>
    <div>
        <h1>Parent</h1>

        <NuxtPage v-slot="{ Component }">
            <template v-if="Component">
                <Suspense>
                    <component :is="Component"></component>

                    <template #fallback>
                        Loading...
                    </template>
                </Suspense>
            </template>
        </NuxtPage>
    </div>
</template>
```

**Child component (nested)**
```
<script lang="ts" setup>
    await new Promise(resolve => setTimeout(resolve, 1000))
</script>

<template>
    <div>
        <h2>Child</h2>
    </div>
</template>

```

**Comment by ol1s** [comment]
Same question from me - how do we customise `fallback` for the `<Suspense>` contained in the `<NuxtPage>` component?

**Comment by ehainer** [comment]
Ditto all the above. This has been one of the pain points as far as understanding how NuxtPage and suspense works.

I don't fully know what, if any the ramifications are for making this change, but replacing `NuxtPage` with `RouterView` instead seems to allow me to use `Suspense` further down the tree. For example, here's my current setup, overly simplified:

app.vue
```
<template>
  <Html>
    <Body>
      <NuxtLayout :name="some-layout-file">
        <RouterView />
      </NuxtLayout>
    </Body>
  </Html>
</template>
```

some-layout-file.vue
```
<template>
  <div class="app">
    <Nav />
    <Masthead :title="title" />
    <slot />
  </div>
</template>
```

The above allows both Nav and Masthead to "persist" through page changes, but within each page (rendered in place of <slot />) I have my Suspense component:

/pages/whatever.vue
```
<template>
  <main class="main">
    <Suspense>
      <!-- All the stuff that is the main "body" of the page that might take a second to load -->
      <SomeSlowComponent />
      <template #fallback>
        <Loader />
      </template>
    </Suspense>
  </main>
</template>
```

In the short while I've been playing with it I seem to be getting the results I want: The masthead and nav don't move/change between page changes, the main body content of the page is replaced with a loading icon if it's not ready yet, and page changes happen immediately (there's no significant delay while it waits for async fetches to finish, it shows the loader instead)

Always happy to be enlightened on the pros/cons of this approach, but, if it ain't broke... right? 🤷🏼‍♂️
